### PR TITLE
Feature/unit test

### PR DESCRIPTION
**Pull request by TimPhillip**
#### Short Description
Adds Unit tests for the NISQ Analyzer.

#### Proposed Changes

  * ImplementationSelectionTest class contains unit tests checking the ability to select a suitable implementation 
  * SimulatorExecutionTest class executes some implementations with the IBMQ_QASM_SIMULATOR qpu and checks the returns. 

**Fixes**: #

### CODE REVIEW

File: org.planqk.nisq.analyzer.core/src/main/java/org/planqk/nisq/analyzer/core/knowledge/prolog/PrologQueryEngine.java [hunk 1]
```diff
@@ -205,7 +206,16 @@ private String assembleQueryForRule(String rule, Map<String, String> params, boo
             }
 
             // FIXME: avoid replacing parts of another variable where the name contains the searched variable, e.g., search for variable 'A' and replace part of variable 'AB' by accident
-            parameterPart = parameterPart.replaceFirst(variable, params.get(variable));
+
+            // Prepare the parameter for Prolog execution
+            String param = params.get(variable).trim();
+            if (!param.matches("-?\\s*[0-9]+\\.?[0-9]*"))
```
Comment (salmma) [inline org.planqk.nisq.analyzer.core/src/main/java/org/planqk/nisq/analyzer/core/knowledge/prolog/PrologQueryEngine.java:212]: Does not work if input is a binary **string**, e.g. 0110. NISQ Analyzer interprets it as an integer

**Review by salmma — CHANGES_REQUESTED** [review]
fix regex handling for Strings

File: org.planqk.nisq.analyzer.core/src/main/java/org/planqk/nisq/analyzer/core/knowledge/prolog/PrologQueryEngine.java (hunk 1 above)
Comment (salmma) [reply org.planqk.nisq.analyzer.core/src/main/java/org/planqk/nisq/analyzer/core/knowledge/prolog/PrologQueryEngine.java:212]: should be deleted!

**Review by salmma — CHANGES_REQUESTED** [review]
Delete old adjustements for inserting '

**Comment by TimPhillip** [comment]
Replaced by #37, because requirements changed over time

**Pull request closed without merging**